--- nim-1.2.2/config/nim.cfg	1592319522.000000000
+++ patches/config/nim.cfg	1592638991.770911502
@@ -317,3 +317,13 @@
   --define:nimOldCaseObjects
   --define:nimOldShiftRight
 @end
+
+@if macosx:
+  @if arm:
+    clang.options.always = "-w -isysroot /usr/share/SDKs/iPhoneOS.sdk -miphoneos-version-min=7.0 -arch armv7 -I/usr/local/include -I/usr/include"
+    clang.options.linker = "-isysroot /usr/share/SDKs/iPhoneOS.sdk -miphoneos-version-min=7.0 -arch armv7 -L/usr/local/lib -L/usr/lib"
+  @elif arm64:
+    clang.options.always = "-w -isysroot /usr/share/SDKs/iPhoneOS.sdk -miphoneos-version-min=7.0 -arch arm64 -I/usr/local/include -I/usr/include"
+    clang.options.linker = "-isysroot /usr/share/SDKs/iPhoneOS.sdk -miphoneos-version-min=7.0 -arch arm64 -L/usr/local/lib -L/usr/lib"
+  @end
+@end
